src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```
@@ -244,6 +244,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <IlcArg Condition="'$(ExportsFile)' != ''" Include="--exportsfile:$(ExportsFile)" />
       <IlcArg Condition="'$(_targetOS)' == 'win' and '$(DebuggerSupport)' != 'false'" Include="--export-dynamic-symbol:DotNetRuntimeDebugHeader,DATA" />
       <IlcArg Condition="'$(_targetOS)' != 'win' and '$(DebuggerSupport)' != 'false'" Include="--export-dynamic-symbol:DotNetRuntimeDebugHeader" />
+      <IlcArg Condition="'$(_targetOS)' == 'freebsd'" Include="--export-dynamic-symbol:__progname;--export-dynamic-symbol:environ" />
```

Do libraries need to re-export these symbols as well or should this be skipped for libraries?

---

Good point. I was actually not testing many variants because i'm running freebsd-x64 on osx-arm64 via qemu with Debug-freebsd-x64 build, which makes it super slow. Just tested a classlib project and found that `lld` (non-fatally) warns:

```
> You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  lib1 succeeded with 2 warning(s) (875.8s) → dist/
    ld.lld : warning : version script assignment of 'V1.0' to symbol '__progname' failed: symbol not defined
    ld.lld : warning : version script assignment of 'V1.0' to symbol 'environ' failed: symbol not defined
```

```suggestion
      <IlcArg Condition="'$(_targetOS)' == 'freebsd' and '$(IsNativeExecutable)' == 'true'" Include="--export-dynamic-symbol:__progname;--export-dynamic-symbol:environ" />
```